test/test_binomial_heap.jl

```diff
@@ -0,0 +1,215 @@
+# Test of binary heaps
+
+# auxiliary functions
+using DataStructures, Test
+
+function heap_values(h::MutableBinomialHeap{VT}) where VT
+    n = length(h)
+    vs = Vector{VT}(undef, n)
+    k = 1
+    val::Union{Nothing,MutableBinomialHeapNode{VT}} = nothing
+    for i = 1:h.cumm_nodecount
+        val = nothing
+        try
+            val = find_node(h, i)
+            vs[k] = val.data
+            k += 1
+        catch _    
```

Comment:
why are we doing this?
Probably we want to be catching an error and then checking if it is the right type
and if it is `continue` and if not `rethrow()`

Catching all exceptions makes debugging harder